snapcraft/cli/lifecycle.py

```diff
@@ -31,6 +40,9 @@ def _execute(command, parts, **kwargs):
     else:
         try:
             lifecycle.execute(command, project_options, parts)
+        except errors.SnapcraftUserPermissionError as user_error:
+            if user_error.user == 'root':
+                _reexec_as_root()
```

Comment:
It's only build that needs this, right? Doesn't this effectively make _every_ lifecycle step run as root? Is that necessary, or can we limit it to build? I suppose everything would be owned by root at that point...